octobot_commons/databases/database_manager.py

```diff
@@ -0,0 +1,136 @@
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import os
+
+import octobot_commons.databases.adaptors as adaptors
+import octobot_commons.constants as constants
+import octobot_commons.symbol_util as symbol_util
+
+
+class DatabaseManager:
+    def __init__(self, tentacle_class, database_adaptor=adaptors.TinyDBAdaptor, backtesting_id=None,
+                 optimizer_id=None, context=None):
+        self.database_adaptor = database_adaptor
+        self.backtesting_id = backtesting_id
+        self.optimizer_id = optimizer_id
+        self.tentacle_class = tentacle_class
+        self.context = context
+        self.base_path = self._merge_parts(constants.USER_FOLDER, tentacle_class.__name__)
+        self.suffix = constants.TINYDB_EXT if self.database_adaptor == adaptors.TinyDBAdaptor else ""
+
+    async def initialize(self, exchange=None):
+        if self.database_adaptor == adaptors.TinyDBAdaptor:
+            deepest_path = self._base_folder() if exchange is None else self._merge_parts(self._base_folder(), exchange)
+            if not os.path.exists(deepest_path):
+                os.makedirs(deepest_path)
+
+    def get_run_data_db_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(), f"{constants.RUN_DATA_DB}{self.suffix}")
+
+    def get_orders_db_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(), f"{constants.ORDERS_DB}{self.suffix}")
+
+    def get_trades_db_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(), f"{constants.TRADES_DB}{self.suffix}")
+
+    def get_symbol_db_identifier(self, exchange, symbol) -> str:
+        return self._merge_parts(self._base_folder(), exchange, f"{symbol_util.merge_symbol(symbol)}{self.suffix}")
+
+    def get_backtesting_metadata_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(ignore_backtesting_id=True), f"{constants.METADATA}{self.suffix}")
+
+    def get_backtesting_run_folder(self) -> str:
+        return self._base_folder()
+
+    def get_optimizer_runs_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(ignore_backtesting_id=True))
+
+    def get_optimizer_runs_schedule_identifier(self) -> str:
+        return self._merge_parts(self.base_path, constants.OPTIMIZER,
+                                 f"{constants.OPTIMIZER_RUNS_SCHEDULE_DB}{self.suffix}")
+
+    async def generate_new_backtesting_id(self) -> int:
+        return await self._generate_new_id(is_optimizer=False)
+
+    async def generate_new_optimizer_id(self, back_list) -> int:
+        return await self._generate_new_id(back_list=back_list, is_optimizer=True)
+
+    async def _generate_new_id(self, back_list=None, is_optimizer=False):
```

Comment:
Would it have been possible to use hashes rather than id that we have to check the uniqueness?

Reply:
already discussed, I'm resolving this
(id is important for user display)